examples/dpsgd_example.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2021 DeepMind Technologies Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""JAX efficiently trains a differentially private conv net on MNIST.
+This script contains a JAX implementation of Differentially Private Stochastic
+Gradient Descent (https://arxiv.org/abs/1607.00133). DPSGD requires clipping
+the per-example parameter gradients, which is non-trivial to implement
+efficiently for convolutional neural networks.  The JAX XLA compiler shines in
+this setting by optimizing the minibatch-vectorized computation for
+convolutional architectures. Train time takes a few seconds per epoch on a
+commodity GPU.
+This code depends on tensorflow_privacy (https://github.com/tensorflow/privacy)
+  Install instructions:
+    $ pip install tensorflow git+https://github.com/tensorflow/privacy.git
+The results match those in the reference TensorFlow baseline implementation:
+  https://github.com/tensorflow/privacy/tree/master/tutorials
+Example invocations:
+  # this non-private baseline should get ~99% acc
+  python -m examples.differentially_private_sgd \
+    --dpsgd=False \
+    --learning_rate=.1 \
+    --epochs=20 \
+   this private baseline should get ~95% acc
+  python -m examples.differentially_private_sgd \
+   --dpsgd=True \
+   --noise_multiplier=1.3 \
+   --l2_norm_clip=1.5 \
+   --epochs=15 \
+   --learning_rate=.25 \
+  # this private baseline should get ~96.6% acc
+  python -m examples.differentially_private_sgd \
+   --dpsgd=True \
+   --noise_multiplier=1.1 \
+   --l2_norm_clip=1.0 \
+   --epochs=60 \
+   --learning_rate=.15 \
+  # this private baseline should get ~97% acc
+  python -m examples.differentially_private_sgd \
+   --dpsgd=True \
+   --noise_multiplier=0.7 \
+   --l2_norm_clip=1.5 \
+   --epochs=45 \
+   --learning_rate=.25 \
+"""
+
+import time
+import warnings
+
+from absl import app
+from absl import flags
+
+import jax
+import jax.numpy as jnp
+from jax.experimental import stax
+import optax
+
+import datasets
```

Comment:
TODO: Change this import to `from optax.examples import datasets` after figuring out how to run the code.